amundsen_application/static/js/components/AvatarLabelList/index.tsx

```diff
@@ -0,0 +1,215 @@
+      this.setState({ itemProps: this.state.tempItemProps, showModal: false });
+    }
+    const onFailureCallback = () => {
+      this.setState({ errorText: DEFAULT_ERROR_TEXT, readOnly: true, showModal: false });
+    }
+    this.props.onUpdateList(updateArray, onSuccessCallback, onFailureCallback);
+  }
+
+  recordAddItem = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    console.log(this.inputRef);
+    const value = this.inputRef.current.value;
+    if (value) {
+      this.inputRef.current.value = '';
+      const newTempItemProps = new Map(this.state.tempItemProps);
+      newTempItemProps.set(value, { label: value });
+      this.setState({ tempItemProps: newTempItemProps });
+    }
+  }
+
+  recordDeleteItem = (key: string) => {
+    const newTempItemProps = new Map(this.state.tempItemProps);
+    newTempItemProps.delete(key)
+    this.setState({ tempItemProps: newTempItemProps });
+  }
+
+  render() {
+    let content;
+    const items = [];
+    const modalItems = [];
+
+    if (this.state.errorText) {
+      return (
+        <div className='editable-list-component'>
+          <label className="status-message">{this.state.errorText}</label>
+        </div>
+      );
+    }
+
+    if (this.state.itemProps.size === 0) {
+      content = (<label className="status-message">No entries exist</label>);
+    }
+    else {
+      this.state.itemProps.forEach((value, key) => {
+        const item = (
+          <li key={`list-item:${key}`}>
+            { React.createElement(AvatarLabel, value) }
+          </li>
+        );
+        items.push(item);
+      });
+
+      content = (
+        <ul className='component-list'>
+          { items }
+        </ul>
+      );
+    }
+
+    if (this.state.showModal) {
+      this.state.tempItemProps.forEach((value, key) => {
+        const modalItem = (
+          <li key={`modal-list-item:${key}`}>
+            { React.createElement(AvatarLabel, value) }
+            <button
+              className='btn delete-button'
+              aria-label='Delete Item'
+              /* tslint:disable - TODO: Investigate jsx-no-lambda rule */
+              onClick={() => this.recordDeleteItem(key)}
+              /* tslint:enable */
+            >
+             <img className='icon icon-delete'/>
+            </button>
+          </li>
+        );
+        modalItems.push(modalItem);
+      });
+    }
+
+    return (
+      <div className='editable-list-component'>
+        { content }
+        {
+          !this.state.readOnly &&
+          <button
+           className='btn add-list-item'
+           onClick={this.handleShow}>
+             <img className='icon icon-plus-circle'/>
+             <span>Add</span>
+          </button>
+        }
+
+        <Modal className='editable-list-modal' show={this.state.showModal} onHide={this.cancelEdit}>
+          <Modal.Header className="text-center" closeButton={false}>
+            <Modal.Title>Owned By</Modal.Title>
```

Comment:
Can "Owned By" and "Enter an email address" be customized? It seems like this component should be generic enough to handle owner editing as well as other editing (select users to share ?). 

If the intent is to only be used for owner editing, then this component should be called "OwnerEditor" or the like.

Reply:
I'll rename it. My intent  for this re-write to worry about making a decent owner editing component, and worry about re-useability when we cross that bridge. That being said, no harm in having these pieces of text also be props.